mods/ts/rules/nod-vehicles.yaml

```diff
@@ -547,3 +547,84 @@ STNK:
 	Explodes@ELITE:
 		RequiresCondition: rank-elite
 		Weapon: UnitExplode
+
+SGEN:
+	Inherits: ^Tank
+	Inherits@VOXELS: ^VoxelActor
+	Inherits@selection: ^SelectableSupportUnit
+	-AppearsOnRadar:
+	Buildable:
+		Queue: Vehicle
+		BuildPaletteOrder: 90
+		Prerequisites: ~naweap, nastlh
+		Description: Can cloak units\nwhen deployed.\n  Unarmed
+	Valued:
+		Cost: 1600
+	Tooltip:
+		Name: Mobile Stealth Generator
+		RequiresCondition: !deployed
+	Tooltip@DEPLOYED:
+		Name: Mobile Stealth Generator (deployed)
+		RequiresCondition: deployed
+	Selectable:
+		Bounds: 1206, 1768, 0, -603
+	Health:
+		HP: 20000
+	Armor:
+		Type: Light
+	Voiced:
+		VoiceSet: Spy
+	Mobile:
+		Speed: 85
+		TurnSpeed: 20
+		ImmovableCondition: !undeployed
+		RequireForceMoveCondition: !undeployed
+	RevealsShroud:
+		RequiresCondition: !inside-tunnel && undeployed
+		Range: 9c0
+		MaxHeightDelta: 3
+	RevealsShroud@DEPLOYED:
+		RequiresCondition: inside-tunnel || deployed
+		Range: 6c0
+		MaxHeightDelta: 3
+	WithMakeAnimation:
+	GrantCondition@PREVIEWWORKAROUND:
+		Condition: real-actor
+	RenderSprites:
+		Image: mstl
+		PlayerPalette: playertem
+	GrantConditionOnDeploy:
+		PauseOnCondition: empdisable || being-captured
+		DeployedCondition: deployed
+		UndeployedCondition: undeployed
+		UndeployOnMove: true
+		UndeployOnPickup: true
+		Facing: 384
+		AllowedTerrainTypes: Clear, Road, DirtRoad, Rough
+		DeploySounds: place2.aud
+		UndeploySounds: clicky1.aud
+		Voice: Move
+	EntersTunnels:
+		RequireForceMoveCondition: !undeployed
+	Repairable:
+		RequireForceMoveCondition: !undeployed
+	Passenger:
```

Comment:
a bunch of these traits should have RequiresCondition: !empdisable

I managed to undeploy an EMP'ed generator by force moving

Reply:
What bunch of traits. Be more specific.